### DISCUSSION TITLE
Idea Statica Detail API

### DISCUSSION THREAD

**Discussion by MadsFriborg** [idea-statica/ideastatica-public#433] [General Discussion]
Hi

Is there currently an API available for the Concrete Detail module?
All I find when looking through different pages is related to connection and rcs.

Regards Mads

**Comment by nathanlukeIDEA** [comment]
Hi @MadsFriborg, we do not have a direct API for Detail. Could you explain the steps you would like to automate?

**Reply by MadsFriborg** [reply]
Hi, the problem I am facing right now is about extracting geometry and sectional forces from larger FEM models (Autodesk Robot, SOFiSTiK or similar), and applying them as boundary conditions in Idea Statica for D-zone calculations.

**Reply by nathanlukeIDEA** [reply]
Hi @MadsFriborg, is it possible to provide an example so we can review it? feel free to email me if you would prefer.

**Comment by nathanlukeIDEA** [comment]
If anyone else is interested in an API for Detail please let us know below with a story on how you would use a detail API?

**Reply by Murtasgh** [reply]
Hi @nathanlukeIDEA, currently we have a simple internal tool that reads the geometry of a RC panel from an existing IFC file and prepares some simple design checks. We are now interested in exporting this information into the Concrete-Detail module to prepare a more accurate design check. Something like the below:

![image](https://github.com/user-attachments/assets/ed6f03c2-1685-4b80-aca8-335ba5ecf388)

With the current version, are there any steps that can be automated?
